docs/developer/tutorials/snmp/profile-format.md

```diff
@@ -0,0 +1,204 @@
+# Profile format reference
+
+## Overview
+
+SNMP profiles are our way of providing out-of-the-box monitoring for certain makes and models of network devices.
+
+An SNMP profile is materialised as a YAML file with the following structure:
+
+```yaml
+sysobjectid: <x.y.z...>
+
+# extends:
+#   <Optional list of base profiles to extend from...>
+
+metrics:
+  # <List of metrics to collect...>
+
+# metric_tags:
+#   <Optional list of tags to apply to collected metrics>
+```
+
+## Fields
+
+### `sysobjectid`
+
+_(Required)_
+
+The `sysobjectid` field is used to match profiles against devices during device autodiscovery.
+
+It can refer to a fully-defined OID for a specific device make and model:
+
+```yaml
+sysobjectid: 1.3.6.1.4.1.232.9.4.10
+```
+
+or a wildcard pattern to address multiple device models:
+
+```yaml
+sysobjectid: 1.3.6.1.131.12.4.*
+```
+
+### `extends`
+
+_(Optional)_
+
+This field can be used to include metrics and metric tags from other so-called _base profiles_. Base profiles can derive from other base profiles to build a hierarchy of reusable profile mixins.
+
+!!! important
+    All device profiles should extend from the `_base.yaml` profile, which defines items that should be collected for all devices.
+
+Example:
+
+```yaml
+extends:
+  - _base.yaml
+  - _generic-router-if.yaml  # Include basic metrics from IF-MIB.
+```
+
+### `metrics`
+
+_(Required)_
+
+Entries in the `metrics` field define which metrics will be collected by the profile. They can reference either a single OID (a.k.a _symbol_), or an SNMP table.
+
+#### Symbol metrics
+
+In profiles, OIDs can be specified as entries containing the `symbol` field:
```

Comment:
Can we make it explicit that symbol metrics from the same MIB still to be separate? As in you can't have a list like:
```
symbol:
  OID: oid1
  name: metric1
   OID: oid2
   name: metric2
```
This was a common mistake I noticed in profiles and the error message is not immediately obvious.